components/tidb_query_expr/src/impl_string.rs

```diff
@@ -499,6 +499,22 @@ pub fn upper(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write_ref(Some(arg)))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower_utf8(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    match str::from_utf8(arg) {
+        Ok(s) => Ok(writer.write_ref(Some(s.to_lowercase().as_bytes()))),
+        Err(err) => Err(box_err!("invalid input value: {:?}", err)),
+    }
+}
+
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    // Noop for binary strings
+    Ok(writer.write_ref(Some(arg)))
```

Comment:
`lower` should behave as current `lower_utf8`. You may substitute body of this function for those in `lower_utf8` and remove `lower_utf8`. Please also add a comment to describe why we implement `lower` different from `upper` and `upper_utf8`.